controllers/predicate.go

```diff
@@ -73,3 +85,14 @@ func (p IgnoreSameContentPredicate) Update(e event.UpdateEvent) bool {
 	}
 	return !reflect.DeepEqual(oldUnstr, newUnstr)
 }
+
+func (p IgnoreSameContentPredicate) handleDeployment(oldObj, newObj *appsv1.Deployment) {
+	// ignore object if only Observed generation changed
+	oldObj.Status.ObservedGeneration = 0
+	newObj.Status.ObservedGeneration = 0
+
+	// ignore annotation which is auto set by Kubernetes
+	revAnnotation := "deployment.kubernetes.io/revision"
```

Comment:
perhaps we should  ignore all annot/labels that have kubernetes.io in them as its reserved for k8s use.